docs/style-guide.rst

```diff
@@ -51,90 +50,558 @@ into the making of Hy.
+                              bararg2
+                              bararg3)  ; aligned with bararg1
+                 arg3)
+
+    (foofunction arg1
+                 (barfunction bararg1
+                   bararg2)  ; Wrong. Looks like a macro body.
+                 arg3)
+
+    (foofunction arg1
+                 (barfunction bararg1 bararg2 bararg3)  ; acceptable.
+                 arg3)
+
+    ;; indenting one space past the parent bracket is acceptable for long lines
+    (foofunction
+      arg1  ; acceptable, but better to keep it on the same line as foofunction
+      (barfunction
+        bararg1  ; indent again
+        bararg2
+        bararg3)
+      arg3)  ; aligned with arg1
+
++ If you need to separate a bracket trail use a ``#_ /`` comment to hold it open.
+  (This avoids violating law #1.)
+
+  .. code-block:: clj
+
+    ;; There are basically two reasons to do this--long lists under version control,
+    ;; and when commenting out the final element during testing.
+    ;; (Common Lisp might use #+(or) for this.)
+
+    ;; preferred
+    [(foo)
+     (bar)
+     (baz)]
+
+    ;; Acceptable if the list is long. (Three isn't that long though.)
+    ;; This is better for version control line diffs.
+    [
+     (foo)
+     (bar)
+     (baz)
+     #_ /]
+
+    ;; Unacceptable and an syntax error. Lost a bracket.
+    [(foo)
+     ;; (bar)
+     ;; (baz)]
+
+    ;; Unacceptable. Broke law #1.
+    [(foo)
+     ;; (bar)
+     ;; (baz)
+     ]
+
+    ;; preferred
+    [(foo)
+     #_(bar)
+     #_(baz)]
+
+    ;; acceptable
+    [(foo)
+     #_
+     (bar)
+     #_
+     (baz)]
+
+    ;; acceptable
+    [(foo)
+     ;; (bar)
+     ;; (baz)
+     #_ /]
+
++ Brackets like to snuggle, don't leave them out in the cold!
+
+  .. code-block:: clj
+
+    ;;; Good
+    [1 2 3]
+    (foo (bar 2))
+
+    ;;; Bad
+    [ 1 2 3 ]
+    ( foo ( bar 2 ) )
+
+    ;;; Ugly
+    [ 1 2 3]
+    (foo( bar 2) )
+
++ Use whitespace to show implicit groups, but be consistent within a form.
+
+  .. code-block:: clj
+
+    ;; Older Lisps would always wrap such groups in even more parentheses.
+    ;; But Hy takes after Clojure, which has a lighter touch.
+
+    {1 9
+     2 8
+     3 7
+     4 6
+     5 5}  ; newlines show key-value pairs in dict
+
+    ;; This grouping makes no sense.
+    #{1 2
+      3 4}  ; It's a set, so why are there pairs?
+
+    ;; This grouping also makes no sense.
+    [1
+     1 2
+     1 2 3]  ; wHy do you like random patterns? [sic pun, sorry]
+
+    ;; BAD. Can't tell key from value without counting
+    {1 9 2 8 3 7 4 6 5 5}
+
+    ;; Good. Extra spaces can work too, if it fits on one line.
+    {1 9  2 8  3 7  4 6  5 5}
+
+    ;; Be consistent. Separate all groups the same way in a form.
+
+    {1 9  2 8
+     3 7  4 6  5 5}  ; Pick one or the other!
+    {1 9  2 8 3 7  4 6  5 5}  ; You forgot something.
+
+    ;; Groups of one must also be consistent.
+
+    (foo 1 2 3}  ; No need for extra spaces here.
```

Comment:
( and }?

Reply:
Wow, I think I need a better font. This was barely visible in my Editor. Switching to Source Code Pro.

Reply:
I'm a fan of DejaVu Sans Mono. Before that, I stuck to Andale Mono.

Reply:
Was using IntelliJ with Python plugins (PyCharm, basically) The one it had before was just called `Monospaced`. I wonder if that's the default. The curlies looked like parens with a little dot.

Source Code Pro is the default for Spacemacs, so I'm used to it.